From 80ba1b4ceca3f15d7e626e4b2fc58cb209b89f63 Mon Sep 17 00:00:00 2001
From: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Date: Wed, 9 Sep 2015 18:55:11 +0300
Subject: [PATCH 103/146] dpaa_eth: fix fq dts parser for macless

This patch converts the BE fq dts entries to the CPU endianness.

Signed-off-by: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
index 68b6e5b..21a3b84 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
@@ -239,7 +239,7 @@ static int dpa_fq_probe_macless(struct device *dev, struct list_head *list,
 				enum port_type ptype)
 {
 	struct device_node *np = dev->of_node;
-	const struct fqid_cell *fqids;
+	struct fqid_cell *fqids;
 	int num_ranges;
 	int i, lenp;
 
@@ -253,6 +253,10 @@ static int dpa_fq_probe_macless(struct device *dev, struct list_head *list,
 
 	/* All ranges defined in the device tree are used as Rx/Tx queues */
 	for (i = 0; i < num_ranges; i++) {
+		/* convert to CPU endianess */
+		fqids[i].start = be32_to_cpup(&fqids[i].start);
+		fqids[i].count = be32_to_cpup(&fqids[i].count);
+
 		if (!dpa_fq_alloc(dev, &fqids[i], list, ptype == RX ?
 				  FQ_TYPE_RX_PCD : FQ_TYPE_TX)) {
 			dev_err(dev, "_dpa_fq_alloc() failed\n");
-- 
2.1.0.27.g96db324

